zmon-controller-app/src/main/java/org/zalando/zmon/controller/GrafanaController.java

```diff
@@ -269,7 +267,7 @@ protected static void replaceVariables(JsonNode node, CheckDefinition checkDefin
         List<CheckDefinition> checkDefinitions = zMonService.getCheckDefinitionsById(checkId);
 
         if (checkDefinitions.isEmpty()) {
-            return new ResponseEntity<JsonNode>(mapper.createObjectNode().put("message", "Check not found"), HttpStatus.NOT_FOUND);
+            return new ResponseEntity<>(mapper.createObjectNode().put("message", "Check not found"), HttpStatus.NOT_FOUND);
```

Comment:
`.put` is for "primitive" types, i.e. non-JsonNode, `.set` is for setting JsonNode children.

`.put` for JsonNode was deprecated, that's why I changed it in some places.

See https://fasterxml.github.io/jackson-databind/javadoc/2.2.0/com/fasterxml/jackson/databind/node/ObjectNode.html